### PR TITLE
ADD: Create CAPPI from Radar

### LINKED ISSUE #597: Constant Altitude Plan Position Indicator (CAPPI)
It would be convenient to create a griddisplay class in line with the radardisplay class to be able to plot CAPPIs in km units.

In addition it may be convenient to be able to plot gridmapdisplay and radarmapdisplay in km units.

### PR DESCRIPTION
**Pull request by syedhamidali**
- [x] Closes #597 #1639 #985 #980 #1585
- [x] Tests added
- [ ] Documentation reflects changes   ---  Not sure

#### Description:
This PR introduces a new feature to generate a Constant Altitude Plan Position Indicator (CAPPI) from radar data directly within Py-ART. The new `create_cappi` function allows users to specify the altitude and radar fields for CAPPI generation, simplifying the process and making it more accessible.

The implementation draws on existing work, with valuable insights from the `comp_z.py` module by @dopplerchase and @vlouf's [CAPPI notebook](https://github.com/vlouf/radar_cappi/blob/master/CAPPI.ipynb). This addition integrates and builds on these concepts, providing a streamlined way to create CAPPI products within Py-ART.

#### Example Usage:

```python
import pyart
import matplotlib.pyplot as plt
from open_radar_data import DATASETS

# Load the sample radar data
file = DATASETS.fetch("RAW_NA_000_125_20080411190016")
radar = pyart.io.read(file)

# Create CAPPI at 2,000 meters and 10,000 meters for the 'reflectivity' field
cappi_2k = pyart.retrieve.create_cappi(radar, fields=['reflectivity'], height=2000)
cappi_10k = pyart.retrieve.create_cappi(radar, fields=['reflectivity'], height=10_000)

# Display the CAPPI at both altitudes
cappi2k_display = pyart.graph.RadarMapDisplay(cappi_2k)
cappi10k_display = pyart.graph.RadarMapDisplay(cappi_10k)

fig, ax = plt.subplots(1, 2, figsize=(12, 4))
cappi2k_display.plot_ppi('reflectivity', vmin=-10, vmax=60, ax=ax[0])
cappi10k_display.plot_ppi('reflectivity', vmin=-10, vmax=60, ax=ax[1])
```
image

This function leverages existing capabilities in Py-ART to offer a consistent and user-friendly method for generating CAPPI visualizations, making it a valuable addition to the toolkit.

### CODE REVIEW

File: pyart/retrieve/cappi.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    fields=None,
+    height=2000,
+    gatefilter=None,
+    same_nyquist=True,
+    nyquist_vector_idx=0,
+):
+    """
+    Create a Constant Altitude Plan Position Indicator (CAPPI) from radar data.
+
+    Parameters:
```
Comment (zssherman) [inline pyart/retrieve/cappi.py:24]: For consistency with other docs can we remove the ':' from the headers

File: pyart/retrieve/cappi.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+    Returns:
+    --------
+    Radar
+        A Py-ART Radar object containing the CAPPI at the specified height.
+
+    Notes:
+    --------
+    CAPPI (Constant Altitude Plan Position Indicator) is a radar visualization
+    technique that provides a horizontal view of meteorological data at a fixed altitude.
```
Comment (zssherman) [inline pyart/retrieve/cappi.py:48]: @mgrover1 Thoughts on this? I'm on the fence with citing wikipedia

**Comment by zssherman** [comment]
@syedhamidali Overall looks good to me, just had two comments

File: pyart/retrieve/cappi.py (hunk 1 above)
Comment (syedhamidali) [reply pyart/retrieve/cappi.py:24]: Done

File: pyart/retrieve/cappi.py (hunk 2 above)
Comment (syedhamidali) [reply pyart/retrieve/cappi.py:48]: Agreed!

Comment (mgrover1) [reply pyart/retrieve/cappi.py:48]: I suggest using the AMS glossary here https://glossary.ametsoc.org/wiki/Cappi

File: pyart/retrieve/cappi.py [hunk 3]
```diff
@@ -37,16 +37,16 @@ def create_cappi(
...
     Radar
         A Py-ART Radar object containing the CAPPI at the specified height.
 
-    Notes:
+    Notes
     --------
     CAPPI (Constant Altitude Plan Position Indicator) is a radar visualization
     technique that provides a horizontal view of meteorological data at a fixed altitude.
-    For more information, see: https://en.wikipedia.org/wiki/Constant_altitude_plan_position_indicator
+    Atlas, David, ed. Radar in Meteorology: Battan Memorial and 40th Anniversary Radar Meteorology Conference. Springer, 2015.
```
Comment (mgrover1) [inline pyart/retrieve/cappi.py:49]: Can you use this one instead? https://glossary.ametsoc.org/wiki/Cappi

**Review by mgrover1 — CHANGES_REQUESTED** [review]
One small suggestion

File: pyart/retrieve/cappi.py (hunk 3 above)
Comment (syedhamidali) [reply pyart/retrieve/cappi.py:49]: Thanks for sharing this!

Comment (mgrover1) [reply pyart/retrieve/cappi.py]: absolutely :) thanks for putting this together!

**Comment by syedhamidali** [comment]
@mgrover1 Done!

**Comment by mgrover1** [comment]
Thanks so much! Looks great to me - nice work @syedhamidali 👍

**Pull request merged**